lib/chef/provider/aws_rds_instance.rb

```diff
@@ -36,14 +36,12 @@ def destroy_aws_object(instance)
     end
     # Wait up to 10 minutes for the db instance to shutdown
     converge_by "waited until RDS instance #{new_resource.name} was deleted" do
-      wait_for(
-        aws_object: instance,
-        query_method: :exists?,
-        expected_responses: [false],
-        acceptable_errors: [AWS::RDS::Errors::DBInstanceNotFound],
-        tries: 60,
-        sleep: 10
-      )
+      tries, sleep = 60, 10
```

Comment:
I addressed this in https://github.com/chef/chef-provisioning-aws/blob/86fda7c7a882dc3fffc3508c5fb8b9d6a6248a26/lib/chef/provider/aws_rds_instance.rb#L39-L48 - I made `wait_for` take an optional block that is called before we execute the query.

I could totally see us modifying `wait_for` to require a block that does the _actual_ checking instead of using the `query_method` parameter.  Thats @jkeiser's suggestion and I agree with it.